pyrsa/util/file_io.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar 25 17:59:29 2020
+
+@author: heiko
+"""
+
+import h5py
+import pickle
+import numpy as np
+
+
+def write_dict_hdf5(file, dictionary):
+    """ writes a nested dictionary containing strings & arrays as data into 
+    a hdf5 file
+
+    Args:
+        file: a filename or opened writable file
+        dictionary(dict): the dict to be saved 
+
+    """
+    file = h5py.File(file, 'a')
+    _write_to_group(file, dictionary)
+    
+    
+def _write_to_group(group, dictionary):
```

Comment:
If the dicts only contain numbers and strings (settings) another option ofcourse is to serialize them with `ast`, or as tuples, or to json. I think this is was nibabel does with parameters in the Nifti header.

Reply:
Yes, I think I will leave this for now. The way I coded it this should be implementable as more functions in this file in the future